src/publisher/middleware.py

```diff
@@ -0,0 +1,107 @@
+import json
+from collections import OrderedDict
+from django.conf import settings
+import logging
+from .utils import lmap
+
+LOG = logging.getLogger(__name__)
+
+# temporarily borrowed from bot-lax-adaptor...
+def visit(data, pred, fn, coll=None):
+    "visits every value in the given data and applies `fn` when `pred` is true "
+    if pred(data):
+        if coll is not None:
+            data = fn(data, coll)
+        else:
+            data = fn(data)
+        # why don't we return here after matching?
+        # the match may contain matches within child elements (lists, dicts)
+        # we want to visit them, too
+    if isinstance(data, OrderedDict):
+        results = OrderedDict()
+        for key, val in data.items():
+            results[key] = visit(val, pred, fn, coll)
+        return results
+    elif isinstance(data, dict):
+        return {key: visit(val, pred, fn, coll) for key, val in data.items()}
+    elif isinstance(data, list):
+        return [visit(row, pred, fn, coll) for row in data]
+    # unsupported type/no further matches
+    return data
+
+#
+#
+#
+
+def downgrade(content):
+    "returns v1-compliant content"
```

Comment:
two key test cases to add:
- `downgrade` should be able to deal with a v2 content (because the bot-lax-adaptor will start producing only those for new articles)
- `upgrade` should be able to deal with with content that is already v2 (same reason, and its `title` field may be missing then)